test/services/xccdf_report_parser_test.rb

```diff
@@ -193,6 +193,24 @@ class TestParser < ::XccdfReportParser
         end
       end
     end
+
+    should 'provide failed results' do
+      @report_parser.save_all
+      failed_states = %w[fail error unknown fixed]
+      failed_rule_results = @report_parser.failed_rule_results
+      assert_equal failed_rule_results.count, 45
+      assert_kind_of ::RuleResult, failed_rule_results.first
```

Comment:
Kind of an anti-pattern in Ruby. What's the justification for checking the type?

Reply:
I know it is not very duck-typeish. Would you suggest something else?

Reply:
I'd personally leave it off completely, but it's not a big deal to leave it either.